### PR TITLE
feat: 536 Add Gitpod config to project

### LINKED ISSUE #536: Add Gitpod config to project
Gitpod is a widely used Open Source Web IDE that aims to provide easy ephemeral dev environments for projects, which are preconfigured with all the setup done and dependencies already installed. Projects making use of it can find that users are able to develop and contribute against the project much more easily.

In order to support Gitpod for community contributions, a simple config file and a little bit of documentation is all that is required. This config file can run the go commands to install and build the project to allow users to get started with writing the code for their changes.

### PR DESCRIPTION
**Pull request by ChevronTango**
Please fill in the fields below to submit a pull request.  The more information that is provided, the better.

Fixes #536

Release Notes: Adds Gitpod config to the project$

**Types of changes**:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). **Please ensure that your PR title** is a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) breaking change (with a `!`, as in `feat!: change foo`). 

**Description of the changes being introduced by the pull request**:

Adds Gitpod config to the project$

**Please verify and check that the pull request fulfills the following requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by rdimitrov — COMMENTED** [review]
My thoughts are that go-tuf is not a complicated project in terms of various services that need to be set up beforehand to run/try it. For the time being, I don't see a lot of value in adding and potentially maintaining a custom config for it. Note that this does not stop anyone from loading go-tuf within a Gitpod workspace. 

I see the benefits of using it, but I don't have a strong "yes" about adding it to the project. I'll let the other maintainers chime in too. 

Disclaimer: I haven't used Gitpod so far so let me know if I'm missing something 👍

**Review by rdimitrov — APPROVED** [review]

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**
